prow/test/integration/test/pod-utils_test.go

```diff
@@ -0,0 +1,622 @@
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Create a Prow Job and make it run.
+			ctx := context.Background()
+
+			// Create a unique podName. It is also used as the name of the
+			// ProwJob CR we will create in the cluster to get a pod (with the
+			// same name) scheduled. Uniqueness is important because we check
+			// to see if this particular ProwJob finished successfully.
+			podName := fmt.Sprintf("test-clonerefs-%s-%s", tt.name, RandomString(t))
+			tt.prowjob.ObjectMeta.Annotations["prow.k8s.io/job"] = podName
+			tt.prowjob.ObjectMeta.Name = podName
+			tt.prowjob.Spec.Job = podName
+
+			// We need to use DecorationConfig, because CloneRefs only gets
+			// called if we have a decorated config.
+			tt.prowjob.Spec.DecorationConfig = &prowjobv1.DecorationConfig{
+				UtilityImages: &prowjobv1.UtilityImages{
+					// These images get created by the integration test.
+					CloneRefs:  "localhost:5001/clonerefs:latest",
+					InitUpload: "localhost:5001/initupload:latest",
+					Entrypoint: "localhost:5001/entrypoint:latest",
+					Sidecar:    "localhost:5001/sidecar:latest",
+				},
+				GCSConfiguration: &prowjobv1.GCSConfiguration{
+					PathStrategy: prowjobv1.PathStrategyExplicit,
+					// This field makes initupload skip the upload to GCS.
+					LocalOutputDir: "/output",
+				},
+			}
+
+			// Set common miscellaneous fields shared by test cases.
+			tt.prowjob.ObjectMeta.Namespace = defaultNamespace
+			tt.prowjob.Spec.Namespace = defaultNamespace
+			tt.prowjob.Spec.Agent = prowjobv1.KubernetesAgent
+			tt.prowjob.Spec.Refs.Org = "some-org"
+			tt.prowjob.Spec.Refs.BaseRef = "master"
+			tt.prowjob.Spec.Refs.WorkDir = true
+			// Use fakegitserver because it ships with sh and git, allowing us
+			// to write self-documenting shell code to run as the test (to see
+			// that clonerefs did actually clone things for us).
+			tt.prowjob.Spec.PodSpec.Containers[0].Image = "localhost:5001/fakegitserver"
+
+			tt.prowjob.Status = prowjobv1.ProwJobStatus{
+				// Setting the state to TriggeredState tells Prow to schedule
+				// (i.e., create a pod) and kick things off.
+				State:     prowjobv1.TriggeredState,
+				StartTime: v1.NewTime(time.Now().Add(-1 * time.Second)),
+			}
+
+			// Set up repos on FGS for just this test case.
+			for _, repoSetup := range tt.repoSetups {
+				buf, err := json.Marshal(repoSetup)
+				if err != nil {
+					t.Fatalf("could not marshal %v", repoSetup)
+				}
+				// Notice that this odd-looking URL is required because we (this
+				// test) is not inside the KIND cluster and so we need to send
+				// the packets to KIND (running on localhost). KIND will then
+				// reroute the packets to fakegitserver.
+				req, err := http.NewRequest("POST", "http://localhost/fakegitserver/setup-repo", bytes.NewBuffer(buf))
+				if err != nil {
+					t.Fatalf("failed to create POST request: %v", err)
+				}
```

Comment:
future reference: this might be used by other tests as well, consider moving this out of `pod-utils_test.go` in the future

Reply:
Done (created a fakegitserver Client which should improve reusability for other tests that would like to talk to the fakegitserver).

I am also tempted to use the common `/cmd/microservicename/main.go` and `/libraryname/library_files.go` pattern to rename the files introduced in this PR, but I'd rather do that refactor in a separate commit.